src/upload.js

```diff
@@ -4,46 +4,115 @@ const FormData = require("form-data");
 const fs = require("fs");
 const p = require("path");
 
-const { defaultOptions, walkDirectory, uriSkynetPrefix } = require("./utils");
-const { SkynetClient } = require("./client");
+const { Upload } = require("tus-js-client");
+
+const { buildRequestHeaders, SkynetClient } = require("./client");
+const { defaultOptions, getFileMimeType, makeUrl, walkDirectory, uriSkynetPrefix } = require("./utils");
+
+/**
+ * The tus chunk size is (4MiB - encryptionOverhead) * dataPieces, set in skyd.
+ */
+const TUS_CHUNK_SIZE = (1 << 22) * 10;
+
+/**
+ * The retry delays, in ms. Data is stored in skyd for up to 20 minutes, so the
+ * total delays should not exceed that length of time.
+ */
+const DEFAULT_TUS_RETRY_DELAYS = [0, 5000, 15000, 60000, 300000, 600000];
 
 const defaultUploadOptions = {
   ...defaultOptions("/skynet/skyfile"),
+  endpointLargeUpload: "/skynet/tus",
+
   portalFileFieldname: "file",
   portalDirectoryFileFieldname: "files[]",
   customFilename: "",
   customDirname: "",
   dryRun: false,
+  largeFileSize: TUS_CHUNK_SIZE,
+  retryDelays: DEFAULT_TUS_RETRY_DELAYS,
 };
 
-SkynetClient.prototype.uploadFile = function (path, customOptions = {}) {
+SkynetClient.prototype.uploadFile = async function (path, customOptions = {}) {
   const opts = { ...defaultUploadOptions, ...this.customOptions, ...customOptions };
 
-  const formData = new FormData();
-  const filename = opts.customFilename ? opts.customFilename : "";
-  formData.append(opts.portalFileFieldname, fs.createReadStream(path), filename);
+  const sizeInBytes = fs.statSync(path).size;
```

Comment:
use async stat, it's already na async function anyway https://nodejs.org/api/fs.html#fs_fspromises_stat_path_options